### PR TITLE
WIP: Voting system

### PR DESCRIPTION
**Pull request by mnapoli**
#37

### CODE REVIEW

File: res/assets/main.js [hunk 1]
```diff
@@ -1,3 +1,41 @@
 $(function () {
-    $('[data-toggle="tooltip"]').tooltip()
+    $('[data-toggle="tooltip"]').tooltip();
+
+    $('body')
+        .on('click', '.upvote', function(e) {
+            e.preventDefault();
+            vote($(this), 1);
+        }).on('click', '.downvote', function(e) {
+        e.preventDefault();
```
Comment (staabm) [inline res/assets/main.js]: Indent

File: res/assets/main.js [hunk 2]
```diff
@@ -1,3 +1,41 @@
...
+        vote($(this), -1);
+    });
 });
+
+function vote($voteLink, value) {
+    if (typeof userId === 'undefined') {
+        alert('You must log in to vote.');
```
Comment (staabm) [inline res/assets/main.js]: Maybe point the user to the login area in this message?

**Comment by staabm** [comment]
Just did some smoke testing.

Why are some arrows golden/yellow even if I did not do any votes yet?

**Comment by mnapoli** [comment]
> Why are some arrows golden/yellow even if I did not do any votes yet?

That's an error then, thanks I'll look into it ;)

**Comment by mnapoli** [comment]
@staabm could you try again when the build gets green again?

File: res/http.php [hunk 3]
```diff
@@ -103,6 +106,23 @@
             ]);
         },
 
+        '/votes/{number}' => route(function (int $number, Voting $voting, ServerRequestInterface $request) {
```
Comment (staabm) [inline res/http.php]: I feel this var could be more explicit -> $emailId

**Comment by staabm** [comment]
> could you try again when the build gets green again

Looks goid, thx.

I added some more notes into the above comment

**Comment by staabm** [comment]
The UI looks really nice, well done!

**Comment by staabm** [comment]
Hmm it doesnt look like the votes do influence the sorting of the threads

**Comment by mnapoli** [comment]
@staabm yes I haven't had time to work on that yet.

**Comment by staabm** [comment]
When space is small or content huge things dont align well
img_5565

**Comment by staabm** [comment]
Looks great!

**Comment by mnapoli** [comment]
Thanks!

**Pull request merged**